### PR TITLE
Lokisk1155/stories backend+redux

### PR DESCRIPTION
**Pull request by lokisk1155**

### CODE REVIEW

File: frontend/src/components/Stories/CreateStory.js [hunk 1]
```diff
@@ -4,31 +4,31 @@ import PreviewStory from "./PreviewStory";
 import TextStory from "./TextStory";
 import { useDispatch, useSelector } from "react-redux";
 import { createStory } from "../../store/story";
+import { Redirect } from "react-router-dom";
```
Comment (dmallon1) [inline frontend/src/components/Stories/CreateStory.js:7]:
```suggestion
```

File: frontend/src/components/Stories/PreviewStory.js [hunk 2]
```diff
@@ -1,15 +1,34 @@
 import { Redirect } from "react-router-dom";
```
Comment (dmallon1) [inline frontend/src/components/Stories/PreviewStory.js:1]:
```suggestion
```

File: frontend/src/components/Stories/PreviewStory.js [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
+import { useHistory } from "react-router-dom";
 
-function PreviewStory({ file, setFile, url, setUrl, submit}) {
+function PreviewStory({ file, setFile, url, setUrl }) {
+  const dispatch = useDispatch();
+  const history = useHistory();
   const simpleUsers = useSelector((state) => state.simpleUsers);
   const sessionUser = useSelector((state) => state.session.user);
+
   const [preview, setPreview] = useState(null);
```
Comment (dmallon1) [inline frontend/src/components/Stories/PreviewStory.js:15]:
```suggestion
```

File: frontend/src/components/Stories/CreateStory.js [hunk 4]
```diff
@@ -114,7 +114,7 @@ function CreateStory() {
       ) : null}
       {photoFile && photoUrl ? (
         <PreviewStory
-        submit={submitStory}
+          submit={submitStory}
```
Comment (dmallon1) [inline frontend/src/components/Stories/CreateStory.js:117]: looks like you're not using this prop in this component

File: frontend/src/routes/StoriesRoutes.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { Switch } from "react-router-dom";
+import { Route } from "react-router-dom";
+import CreateStoryIntro from "../components/Stories/CreateStory";
+
+export function StoriesRoutes() {}
```
Comment (dmallon1) [inline frontend/src/routes/StoriesRoutes.js:5]: do you need this file?

File: frontend/src/store/story.js [hunk 6]
```diff
@@ -8,16 +8,23 @@ export const addStory = (story) => ({
...
-    body: formData instanceof FormData ? formData : JSON.stringify(story)
-  });
-  const storyResponse = await storyRequest.json();
-  if (location === "profile") {
-    return dispatch(profilePage(userId))
-  }
-  return dispatch(addStory(storyResponse));
+export const createStory =
+  (story, userId, location, formData) => async (dispatch) => {
+    const storyRequest = await csrfFetch(`/api/stories?userId=${userId}`, {
```
Comment (dmallon1) [inline frontend/src/store/story.js:13]: you shouldn't need to pass in the current user's id here right?

Comment (lokisk1155) [reply frontend/src/store/story.js]: I am a silly goose

File: frontend/src/routes/StoriesRoutes.js (hunk 5 above)
Comment (lokisk1155) [reply frontend/src/routes/StoriesRoutes.js]: Nah, decided it was unnecessary for now

File: app/controllers/api/stories_controller.rb [hunk 7]
```diff
@@ -1,8 +1,37 @@
...
     def create 
-        @story = Story.new(user_id: params[:user_id].to_i, name: 'this worked')
+        @story = Story.new(user_id: params[:user_id])
+        photo_attached = params[:story] && params[:story].has_key?(:photo)
+        if photo_attached
+            @story.photo.attach(params[:story][:photo])
+        end 
+
+        if !photo_attached
+            @story = false
```
[the remark this answers is not in the document]
Comment (lokisk1155) [reply app/controllers/api/stories_controller.rb:9]: Altered this code

File: app/controllers/api/stories_controller.rb [hunk 8]
```diff
@@ -1,8 +1,37 @@
...
+                end
+            end 
         end
+        @stories = Story.last(3)
+        render 'api/stories/index'
+    end 
+
+    def destroy 
+
     end
```
[the remark this answers is not in the document]
Comment (lokisk1155) [reply app/controllers/api/stories_controller.rb]: no

File: app/controllers/api/stories_controller.rb [hunk 9]
```diff
@@ -1,8 +1,37 @@
...
         if @story.save 
-            render 'api/stories/show'
+            return render 'api/stories/show'
+        end 
+    end 
+
+    def index 
+        @allStories = Story.all 
+        @allStories.each do |story|
+            if story.created_at
```
[the remark this answers is not in the document]
Comment (lokisk1155) [reply app/controllers/api/stories_controller.rb]: just @stories = story.all.last(3) for production for now

**Pull request merged**